crates/oxc_transformer/src/proposals/explicit_resource_management.rs

```diff
@@ -158,10 +158,31 @@ impl<'a> Traverse<'a> for ExplicitResourceManagement<'a, '_> {
     /// ```
     fn exit_static_block(&mut self, block: &mut StaticBlock<'a>, ctx: &mut TraverseCtx<'a>) {
         let scope_id = block.scope_id();
-        if let Some(replacement) =
-            self.transform_statements(&mut block.body, None, scope_id, scope_id, ctx)
+        if let Some((new_stmts, needs_await, using_ctx)) =
+            self.transform_statements(&mut block.body, scope_id, ctx)
         {
-            block.body = ctx.ast.vec1(replacement);
+            let static_block_new_scope_id = ctx.scoping.insert_scope_between(
+                ctx.scoping().scope_parent_id(scope_id).unwrap(),
+                scope_id,
+                ScopeFlags::ClassStaticBlock,
+            );
+
+            ctx.scoping_mut().set_symbol_scope_id(using_ctx.symbol_id, static_block_new_scope_id);
+            ctx.scoping_mut().move_binding(scope_id, static_block_new_scope_id, &using_ctx.name);
+            ctx.scoping_mut().scope_flags_mut(scope_id).remove(ScopeFlags::ClassStaticBlock);
+
+            block.set_scope_id(static_block_new_scope_id);
+            block.body = ctx.ast.vec1(ctx.ast.statement_try(
+                SPAN,
+                ctx.ast.block_statement_with_scope_id(SPAN, new_stmts, scope_id),
+                Some(Self::create_catch_clause(&using_ctx, static_block_new_scope_id, ctx)),
+                Some(Self::create_finally_block(
+                    &using_ctx,
+                    static_block_new_scope_id,
+                    needs_await,
+                    ctx,
+                )),
+            ));
```

Comment:
This logic for creating a `try` statement is repeated in 4 places. Maybe move that logic into a method?

Also, unrelated to this PR, but I just noticed this:

https://github.com/oxc-project/oxc/blob/fd4b0c797467782714221ef6d6db0a1ce3b639c5/crates/oxc_transformer/src/proposals/explicit_resource_management.rs#L629-L631

Seems odd that it's `ctx.current_scope_id()` in one call and `current_scope_id` in the other. Shouldn't they be the same?

Reply:
in the switch stmt, they are the same - let me fix in a different PR.

i'll move the try logic to a fn

Reply:
Thanks. #9966 is merged. Please let me know when this is ready for review again. I'll make it "draft" in meantime.

Reply:
i added it in here > https://github.com/oxc-project/oxc/blob/7dffe96703842d513e741f00ff23b5f9e645e174/crates/oxc_transformer/src/proposals/explicit_resource_management.rs#L724-L731